### PR TITLE
NURBSCurve: Add `toJSON()` and `fromJSON()`.

### LINKED ISSUE #29478: NURBSCurve.toJSON doesn't serialize points data
### Description

The JSON object returned from `NURBSCurve.toJSON()` doesn't contain data relating points, knots or control points, but instead contains:

```
{
    "metadata": {
        "version": 4.6,
        "type": "Curve",
        "generator": "Curve.toJSON"
    },
    "arcLengthDivisions": 200,
    "type": "Curve"
}
```

### Reproduction steps

1. Create an instance of NURBSCurve (sample code below), 
2. `console.log` nurbsCurveInstance.toJSON()
3. Notice there is no data relating to the instance, e.g points, knots and control points)

### Code

```js
const nurbsControlPoints = [];
const nurbsKnots = [];
const nurbsDegree = 3;

for (let i = 0; i <= nurbsDegree; i++) {
  nurbsKnots.push(0);
}

for (let i = 0, j = 20; i < j; i++) {
  nurbsControlPoints.push(
    new THREE.Vector4(
      Math.random(),
      Math.random(),
      Math.random(),
      1 // weight of control point: higher means stronger attraction
    )
  );

  const knot = (i + 1) / (j - nurbsDegree);
  nurbsKnots.push(THREE.MathUtils.clamp(knot, 0, 1));
}

const nurbsCurve = new NURBSCurve(nurbsDegree, nurbsKnots, nurbsControlPoints);

console.log(nurbsCurve.toJSON());
```

### Live example

https://codesandbox.io/p/sandbox/threejs-nurbscurve-tojson-bug-s298gc

### Screenshots

_No response_

### Version

168

### Device

Desktop, Mobile

### Browser

Chrome, Firefox, Safari, Edge

### OS

MacOS

### PR DESCRIPTION
**Pull request by canxerian**
Fixed #29478 

**Description**

Implement `NURBSCurve.toJSON` and `NURBSCurve.fromJSON` methods

Note: `test/unit/src/extras/curves/NURBSCurve.tests.js` added but can be removed (as the corresponding NURBSCurve.js resides in the examples folder).

### CODE REVIEW

File: test/unit/src/extras/curves/NURBSCurve.tests.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/* global QUnit */
+
+import { NURBSCurve } from '../../../../../examples/jsm/curves/NURBSCurve.js';
+import { MathUtils } from '../../../../../src/math/MathUtils.js';
+import { Vector4 } from '../../../../../src/math/Vector4.js';
+
+export default QUnit.module( 'Extras', () => {
```
Comment (Mugen87) [inline test/unit/src/extras/curves/NURBSCurve.tests.js:7]: Do you mind moving this unit test into the `test/unit/addons` folder? The `src` directory is intended for core modules only.

File: test/unit/three.source.unit.js [hunk 2]
```diff
@@ -86,6 +86,7 @@ import './src/extras/curves/LineCurve3.tests.js';
 import './src/extras/curves/QuadraticBezierCurve.tests.js';
 import './src/extras/curves/QuadraticBezierCurve3.tests.js';
 import './src/extras/curves/SplineCurve.tests.js';
+import './src/extras/curves/NURBSCurve.tests.js';
```
Comment (Mugen87) [inline test/unit/three.source.unit.js:89]: Add an entry to `test/unit/three.addons.unit.js` instead.

Comment (canxerian) [reply test/unit/three.source.unit.js]: Done

File: test/unit/src/extras/curves/NURBSCurve.tests.js (hunk 1 above)
Comment (canxerian) [reply test/unit/src/extras/curves/NURBSCurve.tests.js]: Good suggestion, have done so

**Review by Mugen87 — APPROVED** [review]

**Pull request merged**